providers/cncf/kubernetes/src/airflow/providers/cncf/kubernetes/operators/pod.py

```
@@ -935,6 +935,8 @@ def trigger_reentry(self, context: Context, event: dict[str, Any]) -> Any:
             raise
         finally:
             self._clean(event=event, context=context, result=xcom_sidecar_output)
+            if self.do_xcom_push:
+                return xcom_sidecar_output
```

This could be illegal syntax in a future Python version. We'll need to take that `return` out of `finally`.
https://peps.python.org/pep-0765/

---

@AutomationDev85 the more I revisit the core the more I think this patch was not good. Actually you wanted to push XCom but with this you are masking the exception raise.

Can you re-work? I assume you to manually xcom_push() and then raise in case of error, else return XCom in this way only on success but not in finally.

---

@johnslavik thanks for identifying this issue!! I created a PR to fix this issue: https://github.com/apache/airflow/pull/58998

---

I assume cause is that the ruff rule https://docs.astral.sh/ruff/rules/return-in-try-except-finally/ (SIM107) is not enabled in the repo, that _should_ have catched this. Will put this on my bucket list to enable... or if somebody else wants to make a PR I assume this is an important check we are missing atm.

---

> Will put this on my bucket list to enable... or if somebody else wants to make a PR I assume this is an important check we are missing atm.

#59019

---

Cool. Thanks @johnslavik !